llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -37,6 +37,10 @@ static cl::opt<bool> LoopEpilogueAnalysis(
     "aie-loop-epilogue-analysis", cl::init(true),
     cl::desc("[AIE] Perform Loop/Epilogue analysis with loop scheduling"));
 
+static cl::opt<int> MaxExpensiveIterations(
+    "aie-loop-aware-expensive-iterations", cl::init(25),
```

Comment:
Is there a rationale behind this number?

Reply:
Yes and no. I feel anything over 50 is too much, and anything below 10 is not enough if we need to move a couple of instructions up by 2-3 cycles. So 25 felt like a good compromise. And this works well for loops with an II between 5 and 10 cycles, which is the territory of the PreRA pipeliner for us.